micrometer-core/src/main/java/io/micrometer/core/instrument/HighCardinalityTagsDetector.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import io.micrometer.common.util.internal.logging.InternalLogger;
+import io.micrometer.common.util.internal.logging.InternalLoggerFactory;
+
+/**
+ * Tries to detect high cardinality tags by checking if the amount of Meters with the same
+ * name is above a threshold. You can use this class in two ways: 1. Call findFirst and
+ * check if you get any results, if so you probably have high cardinality tags 2. Call
+ * start which will start a scheduled job that will do this check for you
```

Comment:
I think it is worth mentioning `MeterFilter.maximumAllowableMetrics` and `MeterFilter.maximumAllowableTags` as related tools to deal with having too many meters.

Reply:
👍🏼